addons/account/models/account_move.py

```diff
@@ -3037,6 +3030,26 @@ def _get_update_invoice_from_attachment_decoders(self, invoice):
         """
         return []
 
+    @api.model
+    def _get_company_terms(self, company):
+        if not company.terms_type == 'html':
+            return company.invoice_terms if not is_html_empty(company.invoice_terms) else ''
+        baseurl = self.env.company.get_base_url() + '/terms'
+        return _('Terms & Conditions: %s', baseurl)
+
+    @api.depends('move_type', 'partner_id')
+    def _compute_narration(self):
+        use_invoice_terms = self.env['ir.config_parameter'].sudo().get_param('account.use_invoice_terms')
+        # Avoid using the default TC for vender bills
+        is_sale = self.env.context.get('default_move_type') in self.env['account.move'].get_sale_types()
```

Comment:
I don't get why you are depending on the context. Seems like a bad idea for a stored field in general, and useless in this case since `move_type` is set on the move.